client/gutenberg/extensions/slideshow/create-swiper.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import Swiper from 'swiper';
+import 'swiper/dist/css/swiper.min.css';
+import { merge } from 'lodash';
+
+export default function createSwiper( container = '.swiper-container', params = {} ) {
+	const defaultParams = {
```

Comment:
Yeah, one of the chief reasons why I removed it was that I couldn't carry it over to the frontend side if I were to drop React there. So if we want to re-implement it, I agree that it's probably best to stick to Swiper's own functionality. In addition to the events you pointed out, I've also found `observerUpdate` which is apparently triggered upon any DOM changes (if `observe` is set to `true`).

I'll play a bit with that (I'll first need to try your original PR on a set of images with reasonably diverse sizes and aspect rations to see its behavior, which I haven't really done so far).

As for communicating the first image's dimensions to the frontend -- TBH I'd be inclined to gather that from the DOM in this case. It seems a bit redundant to pass it as `data-` attrs (which after all we can also only obtain through the DOM, albeit easier), and too much of a detail specific to the slideshow's behavior to store it as part of the block's attributes IMO.

Reply:
Update: This is a tricky one :slightly_smiling_face: 

I tried playing with this a bit. Among the more promising options I found:

- Setting `setWrapperSize: false` (which is actually its default -- Swiper docs say

> Enabled this option and plugin will set width/height on swiper wrapper equal to total size of all slides. Mostly should be used as compatibility fallback option for browser that don't support flexbox layout well.

- Setting `autoHeight: true`. This does at least something to accomodate for different slide sizes -- it changes the Swiper container size accordingly. However, for an image that hasn't finished loading yet, UX is pretty bad since it'll only show a slide with minimal height.

Note that I tried both of these while commenting out our own `style.scss` entirely in order not to interfere.

- With the styling enabled back on, I also tried

```scss
	.swiper-container {
		align-items: center;
	}
```

This is also _somewhat_ promsing, since for a swiper whose first image is landscape, and a later one is portrait, it'll center the landscape ones vertically -- however, only after the first portrait one has been hit. Unfortunately, the corresponding main axis property, `justify-content`, doesn't work here because... uh I think because of the way Swiper uses Flexbox. Maybe it's possible to get it to work (might need `margin: auto` somewhere?)

---

I wouldn't rule out that a CSS only solution is possible (if what we want is keep the first slide's size and fit all subsequent slides into that frame), since  Swiper seems to be heavily using Flexbox for styling.

This MDN article might have helpful information: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Controlling_Ratios_of_Flex_Items_Along_the_Main_Ax

Reply:
I :heart: #30438